### PR TITLE
fix(comick): genres not loading

### LINKED ISSUE #530: ComicK missing manga metadata
https://github.com/Skittyblock/aidoku-community-sources/blob/3a95a55bcd00c2f1c3e860eafc33dbfea027f6ff/src/rust/multi.comick/src/parser.rs#L221C6-L221C6

When parsing the listing, only the title is loaded, this ends up with all of the comick manga being empty. No further information available even after refreshing metadata.

### PR DESCRIPTION
**Pull request by daarkdemon**
Checklist:
- [x] Updated source's version for individual source changes
- [ ] Updated all sources' versions for template changes
- [ ] Set appropriate `nsfw` value
- [ ] Did not change `id` even if a source's name or language were changed
- [x] Tested the modifications by running it on the simulator or a test device 

Fixes #530

### CODE REVIEW

File: src/rust/multi.comick/src/parser.rs [hunk 1]
```diff
@@ -148,7 +148,7 @@ pub fn parse_manga_list(
 		}
 		Ok(MangaPageResult {
 			manga: mangas,
-			has_more: false,
+			has_more: true,
```
Comment (Skittyblock) [inline src/rust/multi.comick/src/parser.rs:151]: this shouldn't always be true, otherwise it'll loop infinitely. you need to have a check (e.g. if `mangas` is empty) to see if there are more or not.

**Review by Skittyblock — CHANGES_REQUESTED** [review]
in the future I would probably enforce not using `expect`, since it should only ever be used if you are fully certain it should never be triggered, and since sites change so much it can happen, but it's fine for now until 0.7.

File: src/rust/multi.comick/src/parser.rs (hunk 1 above)
Comment (daarkdemon) [reply src/rust/multi.comick/src/parser.rs]: fiixed please check now

File: src/rust/multi.comick/src/lib.rs [hunk 2]
```diff
@@ -10,8 +10,8 @@ use aidoku::{
 pub mod helper;
 pub mod parser;
 
-const BASE_URL: &str = "https://comick.cc";
-const API_URL: &str = "https://api.comick.cc";
+const BASE_URL: &str = "https://comick.fun";
+const API_URL: &str = "https://api.comick.fun";
```
Comment (Skittyblock) [inline src/rust/multi.comick/src/lib.rs:14]: why did you change back to the old `.fun` domain? `.cc` is the new one

File: src/rust/multi.comick/src/parser.rs [hunk 3]
```diff
@@ -106,12 +106,12 @@ pub fn parse_manga_list(
...
 		);
 		let mut mangas: Vec<Manga> = Vec::new();
-		let json = Request::new(&url, HttpMethod::Get)
+		let json = Request::new(url, HttpMethod::Get)
 			.json()
 			.expect("Failed to load JSON")
 			.as_array()
 			.expect("Failed to get json as array");
-		for data in json {
+		for data in json.clone() {
```
[the remark this answers is not in the document]
Comment (daarkdemon) [reply src/rust/multi.comick/src/parser.rs:114]: Still the clone() will be used for that variable ?

File: src/rust/multi.comick/src/lib.rs (hunk 2 above)
Comment (daarkdemon) [reply src/rust/multi.comick/src/lib.rs:14]: It is mentioned in API docs to use  `.fun` for api and `.cc` for website

image

File: src/rust/multi.comick/src/parser.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (daarkdemon) [reply src/rust/multi.comick/src/parser.rs]: Resolved

File: src/rust/multi.comick/src/lib.rs (hunk 2 above)
Comment (Skittyblock) [reply src/rust/multi.comick/src/lib.rs:14]: shouldn't `BASE_URL` be `.cc`?

**Comment by daarkdemon** [comment]
@Skittyblock any update?

File: src/rust/multi.comick/src/lib.rs [hunk 4]
```diff
@@ -11,7 +11,7 @@ pub mod helper;
 pub mod parser;
 
 const BASE_URL: &str = "https://comick.cc";
```
Comment (Skittyblock) [inline src/rust/multi.comick/src/lib.rs:13]: #598 
```suggestion
const BASE_URL: &str = "https://comick.io";
```

**Review by Skittyblock — APPROVED** [review]
